src/integrationTest/java/uk/gov/hmcts/darts/stub/services/DownloadProxyTest.java

```diff
@@ -0,0 +1,119 @@
+        assertAllOtherThanClientIdAreSame(CsvUtil.toDownloadCsvRecord(CsvUtil.getExpectedData()).get(0), records.get(1));
+        assertAllOtherThanClientIdAreSame(CsvUtil.toDownloadCsvRecord(CsvUtil.getExpectedData()).get(0), records.get(2));
+    }
+
+    @Test
+    void testDownloadWithOneEod() throws Exception {
+        MultiValueMap<String, String> headers = new HttpHeaders();
+        headers.put("Content-Type", List.of("application/json"));
+        headers.put(EodRequestWrapper.EOD_REQUEST_HEADER, List.of("100"));
+        HttpEntity<String> entity = new HttpEntity<>("", headers);
+
+        ResponseEntity<String> result = this.restTemplate.exchange(uri, HttpMethod.GET, entity, String.class);
+        assertThat("application/csv").isEqualTo(result.getHeaders().get("Content-Type").get(0));
+        List<DownloadCsvRecord> records = CsvUtil.toDownloadCsvRecord(result.getBody());
+        Assertions.assertEquals(1, records.size());
+        Assertions.assertEquals("100", records.get(0).getClientIdentifier());
+        Assertions.assertFalse(anyNull(records.get(0)));
+        assertAllOtherThanClientIdAreSame(CsvUtil.toDownloadCsvRecord(CsvUtil.getExpectedData()).get(0), records.get(0));
+    }
+
+    @Test
+    void testDownloadWithNoEod() throws Exception {
+        MultiValueMap<String, String> headers = new HttpHeaders();
+        headers.put("Content-Type", List.of("application/json"));
+        HttpEntity<String> entity = new HttpEntity<>("", headers);
+
+        ResponseEntity<String> result = this.restTemplate.exchange(uri, HttpMethod.GET, entity, String.class);
+        assertThat("application/csv").isEqualTo(result.getHeaders().get("Content-Type").get(0));
+        List<DownloadCsvRecord> records = CsvUtil.toDownloadCsvRecord(result.getBody());
+        Assertions.assertEquals(1, records.size());
+        Assertions.assertEquals("Client Identifier", records.get(0).getClientIdentifier());
+        Assertions.assertFalse(anyNull(records.get(0)));
+        Assertions.assertEquals(CsvUtil.toDownloadCsvRecord(CsvUtil.getExpectedData()).get(0), records.get(0));
+    }
+
+    private void assertAllOtherThanClientIdAreSame(DownloadCsvRecord expected, DownloadCsvRecord actual) {
+        expected.setClientIdentifier("Client Identifier");
+        actual.setClientIdentifier("Client Identifier");
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @SuppressWarnings("PMD.AvoidAccessibilityAlteration")
+    public static boolean anyNull(Object target) {
+        return Arrays.stream(target.getClass()
+                                 .getDeclaredFields())
+            .peek(f -> f.setAccessible(true))
+            .map(f -> getFieldValue(f, target))
+            .anyMatch(Objects::isNull);
+    }
+
+    private static Object getFieldValue(Field field, Object target) {
+        try {
+            return field.get(target);
+        } catch (IllegalAccessException e) {
+            throw new AssertionError(e);
+        }
+    }
```

Comment:
As this always validates the same object would be better to use the object default getters / fields instead of using reflections

Reply:
Can do, just wanted to make a quick pass over the object. I will change

Reply:
Looking at this again the lombok equals should do the logical equality for us e.g.      Assertions.assertEquals(CsvUtil.toDownloadCsvRecord(CsvUtil.getExpectedData()).get(0), records.get(0));
I have removed the null checks